### ISSUE TITLE
Add `preDeleteCommand` to devcontainer.json

### ISSUE THREAD

**Issue by chrmarti** [microsoft/vscode-remote-release#2985]
@Chuxel Moving the `preDeleteCommand` discussion here. I realize that this might be mainly driven by Codespaces, we just need to understand if/how it applies to Remote-Containers.

`preDeleteCommand` needs more discussion:

The only case where Remote-Containers deletes a container automatically is when rebuilding the container. Should it run the `preDeleteCommand` in that case? The container will be recreated after that.

The user can delete a container through the Remote Explorer. Should the `preDeleteCommand` run before that?

What if the user deletes the container through the Docker extension or some other tool or the Docker CLI? Is not running the configured command in this case good enough?

_Originally posted by @chrmarti in https://github.com/microsoft/vscode-remote-release/issues/2772#issuecomment-627372461_

**Comment by Chuxel** [comment]
Yeah this one is a bit tricky like you say. It would have to be documented as "best effort" and that we recommend you use the remote explorer or commands in the extension to ensure the command is fired.

To help a bit, in other cases, one thing we could potentially do for others is track known devcontainer.json created containers with a `preDeleteCommand` by ID.  Any time a container for the same devcontainer.json starts, we see if there is an ID that does not match the container being started or created (if it exists). If one is found, we prompt as to whether the `preDeleteCommand` command in the container should be run before executing other commands.

The only challenge is that this would run in the new container instead of the old, now deleted one - which is why I think we'd want to prompt.

**Comment by chrmarti** [comment]
Also waiting for internal feedback (https://github.com/microsoft/vscode-internalbacklog/issues/909#issuecomment-630950985).
